comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -420,6 +423,12 @@ func (r *secretResolver) Refresh() (string, error) {
 
 	// when Refreshing secrets, only update what the allowlist allows by passing `true`
 	refreshResult := r.processSecretResponse(secretResponse, true)
+	// add the results to the audit file, if filename is non-empty
+	if r.auditFilename != "" {
+		if err := AddToRefreshAuditFile(r.auditFilename, secretResponse, r.origin); err != nil {
+			return "", err
+		}
+	}
```

Comment:
Depending on the configuration of the refresh interval we might writing to disk very often. I believe that is not ideal 😄 

What is we implement `AddToRefreshAuditFile` using a buffer, and only write to disk when is full or about to get full? 

It would probably make the implementation a bit more complex but it might yield better performance overall for the agent.

Reply:
I think detecting when a secret value changed, ie different from what we have in the cache, would solve the issue without making the code more complex.
Also we, in case of a panic we don't want some audit log to be lost.

Reply:
This was a mistake on my part, it should only be adding to the audit file when secrets actually change, which we anticipate won't happen very often. This also explains why it's not a problem to unmarshal and remarshal json every time.